docs/source/advanced.rst

```diff
@@ -0,0 +1,354 @@
+.. role:: python(code)
+   :language: python
+
+Advanced
+========
+
+.. _init-model-factory:
+
+:samp:`init_model_factory`
+--------------------------
+
+The :samp:`init_model_factory` interface is less user friendly but perhaps of
+interest to advanced users. It accepts the specification in dictionary format
+(so it has fewer dependencies than :ref:`init-yaml` and :ref:`init-json`) and
+does not construct a declarative base. It accepts the following parameters:
+
+* :samp:`base`: The SQLAlchemy declarative base as a keyword only argument. It
+  is used to as the base class for all SQLAlchemy models.
+* :samp:`spec`: The OpenAPI specification as a dictionary as a keyword only
+  argument.
+* :samp:`models_filename`: The name of the file where the SQLAlchemy models
+  will be written as an optional keyword only argument.
+
+.. note:: the :samp:`define_all` parameter has been removed and OpenAlchemy
+  behaves as though it is set to :samp:`True`.
+
+The return value is the :samp:`model_factory` as defined as part of the return
+value of :ref:`init-yaml`.
+
+.. _build-yaml:
+
+:samp:`build_yaml`
+------------------
+
+Used to build a package with the SQLAlchemy models (including type hints) based
+on a YAML OpenAPI specification which has been extended with any relevant
+OpenAlchemy extension properties.
+
+The :samp:`build_yaml` interface requires the :samp:`PyYAML` library to be
```

Comment:
Again not specifically related to this PR but something I noticed, do you think there should be a line here that shows how to install with yaml? I.e. `pip install OpenAlchemy[yaml]` I believe

Reply:
I can add it here if you really want to, but we already mention that in the `README.rst` file (in the installation section), so I think the information would be redundant.

Reply:
Fair point :smile: